bloom-housing/bloom
# App csv performance 2

## Issue Overview

This PR addresses #2020 


## Description

This overhauls how we fetch applications for the csv export and how we transform that into a csv. By updating it to use getRawMany and directly transform the result into a csv, we are able to gain an additional 5x in speed from the last round of updates. To put that in perspective, by seeding my local db with 10k applications per listing, I am now able to export 10k in ~5 seconds total, where what was originally there would take 60+ seconds and then we got it down to ~25 seconds with the same amount of applications.

This also cleans up the fields that are exported, as outlined in the issue.

## How Can This Be Tested/Reviewed?

Run the export and you should see an export with the fields as mapped in `application-csv-exporter`

I still have to update tests, but we can get a head start on review.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
